Error with py.test .

Please describe your issue here:

Hello,

I am having error of testing ShakeMap 4. The installation step did appear correct. I did try git pull and reinstall.

The following is the error message.

---
==================================== ERRORS ====================================
____________ ERROR collecting tests/shakemap/coremods/dyfi_test.py _____________
ImportError while importing test module '/Users/liliye/shakemap_src/tests/shakemap/coremods/dyfi_test.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
../opt/anaconda3/envs/shakemap/lib/python3.8/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
tests/shakemap/coremods/dyfi_test.py:9: in <module>
    from shakemap.coremods.dyfi import _get_dyfi_dataframe
shakemap/coremods/dyfi.py:9: in <module>
    from impactutils.io.table import dataframe_to_xml
E   ModuleNotFoundError: No module named 'impactutils.io.table'
-------- generated xml file: /Users/liliye/shakemap_src/test-output.xml --------
=========================== short test summary info ============================
ERROR tests/shakemap/coremods/dyfi_test.py
!!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!
=============================== 1 error in 1.90s ===============================

I run the code in shakemap environment. It displayed the same type error



(shakemap) liliye@liliyes-MacBook-Air shakemap_src % pwd
/Users/liliye/shakemap_src
(shakemap) liliye@liliyes-MacBook-Air shakemap_src % py.test .
============================= test session starts ==============================
platform darwin -- Python 3.8.5, pytest-6.2.1, py-1.10.0, pluggy-0.13.1
rootdir: /Users/liliye/shakemap_src, configfile: pytest.ini
plugins: azurepipelines-0.8.0, cov-2.11.1
collected 134 items / 1 error / 133 selected                                   
##vso[results.publish type=JUnit;runTitle='Pytest results';]/Users/liliye/shakemap_src/test-output.xml
##vso[task.logissue type=error;]1 test(s) failed, 134 test(s) collected.
##vso[task.logissue type=warning;]Coverage XML was not created, skipping upload.

==================================== ERRORS ====================================
____________ ERROR collecting tests/shakemap/coremods/dyfi_test.py _____________
ImportError while importing test module '/Users/liliye/shakemap_src/tests/shakemap/coremods/dyfi_test.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
../opt/anaconda3/envs/shakemap/lib/python3.8/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
tests/shakemap/coremods/dyfi_test.py:9: in <module>
    from shakemap.coremods.dyfi import _get_dyfi_dataframe
shakemap/coremods/dyfi.py:9: in <module>
    from impactutils.io.table import dataframe_to_xml
E   ModuleNotFoundError: No module named 'impactutils.io.table'
-------- generated xml file: /Users/liliye/shakemap_src/test-output.xml --------
=========================== short test summary info ============================
ERROR tests/shakemap/coremods/dyfi_test.py
!!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!
=============================== 1 error in 3.62s ===============================
(shakemap) liliye@liliyes-MacBook-Air shakemap_src % conda info -envs
usage: conda info [-h] [--json] [-v] [-q] [-a] [--base] [-e] [-s]
                  [--unsafe-channels]
conda info: error: argument -e/--envs: ignored explicit argument 'nvs'
(shakemap) liliye@liliyes-MacBook-Air shakemap_src % conda info --envs
# conda environments:
#
base                     /Users/liliye/opt/anaconda3
gmprocess                /Users/liliye/opt/anaconda3/envs/gmprocess
shakemap              *  /Users/liliye/opt/anaconda3/envs/shakemap

--- a/install.sh
+++ b/install.sh
@@ -148,7 +148,7 @@ package_list=(
       "fiona"
       "gdal"
       "h5py"
-      "impactutils"
+      "impactutils=0.8.24"
       "ipython"
       "libcomcat"
       "lockfile"

